Enhance and update documentation (#351)
kubernetes-sigs/kueue: README.md

a job should be [admitted](docs/concepts#admission) to start (as in pods can be
created) and when it should stop (as in active pods should be deleted).

## Why use Kueue

Kueue is a lean controller that you can install on top of a vanilla Kubernetes

> cluster. Kue does not replace any existing Kubernetes components. Kueue is

Reviewer: ```suggestion
cluster. Kueue does not replace any existing Kubernetes components. Kueue is
```

Author: Ah, how did that happen?
Although not a bad name (it's basically pronounced the same).

a job should be [admitted](docs/concepts#admission) to start (as in pods can be
created) and when it should stop (as in active pods should be deleted).

## Why use Kueue

> Kueue is a lean controller that you can install on top of a vanilla Kubernetes
> cluster without replacing any components. It is compatible with cloud
> environments where:
> - Nodes and other compute resources can be scaled up and down.
> - Compute resources are heterogeneous (in architecture, availability, price, etc.).

Reviewer: ```suggestion
Kueue is a lean controller that you can install on top of a vanilla Kubernetes
cluster. It does not replace any existing Kubernetes components. It is compatible with cloud
environments where:
- Compute resources are elastic and can be scaled up and down.
- Compute resources are heterogeneous (in architecture, availability, price, etc.).
```

Author: Done

Reviewer 2: What does "heterogeneous" mean in this example?

Author: It is compatible, as homogeneous is just simpler. In kueue terms, if your resources are homegeneous, you need only one ResourceFlavor.

Kueue APIs allow you to express:
- Quotas and policies for fair sharing among tenants.

> - Resource fungibility: if a [resource flavor](docs/concepts/cluster_queue.md#resourceflavor-object)

Author: I added the subject. It's kueue that runs the job in a different flavor. I think we don't need the link to workload again. It's in the first paragraph.

- Kueue [concepts](docs/concepts).
- Common and advanced [tasks](docs/tasks).

## Architecture

<!-- TODO(#64) Remove links to google docs once the contents have been migrated to this repo -->

> Learn more about the architecture of Kueue in the design docs:

Author: Done

- [bit.ly/kueue-apis](https://bit.ly/kueue-apis) (please join the [mailing list](https://groups.google.com/a/kubernetes.io/g/wg-batch)
to get access) discusses the API proposal and a high-level description of how it

> operates.

Reviewer: I would keep the `bit.ly/` prefix, sometimes people highlight and want to copy the short link
